plugins/plugin-core-impl/src/connection.ts

```diff
@@ -53,15 +60,21 @@ export async function connect (
   }
   const processor = new RequestProcessor(async () => {
     const s = await doConnect()
-    return {
-      send: (request) => {
-        s.send(serialize(request))
-      }
-    }
-  }, handler)
+    return { send: (req) => s.send(serialize(req)) }
+  })
```

Comment:
I do not get this complication, why not `new RequestProcessor((req) => s.send(serialize(req)))`?

Reply:
Refactored.